sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -365,7 +394,7 @@ public OutputStream newOutputStream(Path path, OpenOption... options) throws IOE
             StandardOpenOption.WRITE,
             // Though we don't actually truncate, the same result is achieved by overwriting the destination.
             StandardOpenOption.TRUNCATE_EXISTING);
-        for (OpenOption option : optionsList) {
+        for (OpenOption option : optionsSet) {
             if (!supportedOptions.contains(option)) {
                 throw LoggingUtility.logError(logger, new UnsupportedOperationException("Unsupported option: "
```

Comment:
In this case should we be throwing `UnsupportedOperationException` or `IllegalArgumentException`? The operation is supported but the parameters passed are invalid which makes me lean towards `IllegalArguementException`.

Reply:
The javadocs for this type say:
IllegalArgumentException - if the set contains an invalid combination of options
UnsupportedOperationException - if an unsupported open option is specified or the array contains attributes that cannot be set atomically when creating the file

I think if the option is not in our set of supported options as in this case it should be UnsupportedOperationException. 

We do an IllegalArgument check a little bit further down: 
```
if (!optionsSet.contains(StandardOpenOption.WRITE)
            || !(optionsSet.contains(StandardOpenOption.TRUNCATE_EXISTING)
            || optionsSet.contains(StandardOpenOption.CREATE_NEW))) {
            throw LoggingUtility.logError(logger,
                new IllegalArgumentException("Write and either CreateNew or TruncateExisting must be specified to open "
                    + "an OutputStream"));
        }
```

Lmk what you think